cmd/charmd/main.go

```diff
@@ -123,15 +122,25 @@ func serve(conf *config.Config) error {
 	case config.MongoDBBlobStore:
 		// This is the default. No need for a custom function.
 	case config.SwiftBlobStore:
-		cred := &identity.Credentials{
-			URL:        conf.SwiftAuthURL,
-			User:       conf.SwiftUsername,
-			Secrets:    conf.SwiftSecret,
-			Region:     conf.SwiftRegion,
-			TenantName: conf.SwiftTenant,
+		authVersion := 0
+		switch conf.SwiftAuthMode {
+		case "legacy":
+			authVersion = 1
+		case "userpass":
+			authVersion = 2
+		case "userpassv3":
+			authVersion = 3
```

Comment:
maybe add a default case here?

Reply:
Also, given that these are fairly arbitrary constants, have we got sufficient test coverage to be sure they're actually correct?

Reply:
In the default case it is autodetected.